libbeat/processors/annotate/kubernetes/indexing.go

```diff
@@ -167,28 +173,90 @@ func (m *Matchers) MetadataIndex(event common.MapStr) string {
 	return ""
 }
 
+type GenDefaultMeta struct{}
+
 // GenMetadata generates default metadata for the given pod
-func GenMetadata(pod *corev1.Pod) common.MapStr {
+func (g *GenDefaultMeta) GenerateMetaData(pod *corev1.Pod) common.MapStr {
 	labelMap := common.MapStr{}
 	for k, v := range pod.Metadata.Labels {
 		labelMap[k] = v
 	}
-	return common.MapStr{
+
+	meta := common.MapStr{
 		"pod":       pod.Metadata.GetName(),
 		"namespace": pod.Metadata.GetNamespace(),
-		"labels":    labelMap,
 	}
+
+	if len(labelMap) != 0 {
+		meta["labels"] = labelMap
+	}
+
+	return meta
+}
+
+type GenFilteredMeta struct {
```

Comment:
I'm wondering, why do you need 2 different metadata generators? All the filtering logic could go into the default, isn't it?